internal/api/ratelimits.go

```diff
@@ -0,0 +1,46 @@
+package api
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/supabase/auth/internal/conf"
+)
+
+type SimpleRateLimiter struct {
+	Rate conf.Rate
+
+	counter uint64
+}
+
+func NewSimpleRateLimiter(now time.Time, rate conf.Rate) *SimpleRateLimiter {
+	r := &SimpleRateLimiter{
+		Rate: rate,
+	}
+
+	counterStartedAt := now.Truncate(rate.OverTime)
+	counterResetsAt := counterStartedAt.Add(rate.OverTime)
+
+	proRate := float64(counterStartedAt.Sub(now).Milliseconds()) / float64(rate.OverTime.Milliseconds())
+
+	r.counter = uint64(rate.Events * proRate)
+
+	go r.fillBucket(counterResetsAt.Sub(now))
+
+	return nil
+}
+
+func (r *SimpleRateLimiter) Increment(events uint64) bool {
+	return atomic.AddUint64(&r.counter, events) < uint64(r.Rate.Events)
+}
+
+func (r *SimpleRateLimiter) fillBucket(initialWait time.Duration) {
```

Comment:
hmm this function is basically counting down to the bucket expiry and resetting the counter right? should we name this something else instead? naming it `fillBucket` confused me a little and i thought this was where we incremented the counter